libraries/plugins/tags/include/scorum/tags/tags_api_impl.hpp

```diff
@@ -433,6 +433,49 @@ class tags_api_impl
         return result;
     }
 
+    posts_crefs exclude(const posts_crefs& filtering_posts,
+                        const discussion_query& query,
+                        const std::function<bool(const tag_object&)>& tag_filter) const
+    {
+        auto rng_exclude = query.exclude_tags | boost::adaptors::transformed(utils::to_lower_copy)
+            | boost::adaptors::transformed([](const std::string& s) { return utils::substring(s, 0, TAG_LENGTH_MAX); });
+
+        std::set<std::string> tags_exclude(rng_exclude.begin(), rng_exclude.end());
+        if (tags_exclude.empty())
+            return filtering_posts;
+
+        std::vector<posts_crefs> exclude_posts_by_tags;
+        exclude_posts_by_tags.reserve(tags_exclude.size());
+
+        boost::transform(tags_exclude, std::back_inserter(exclude_posts_by_tags),
+                         [&](const std::string& t) { return get_posts(t, tag_filter); });
+
+        if (exclude_posts_by_tags.empty())
+            return filtering_posts;
+
+        using post_cref = posts_crefs::value_type;
+
+        struct less_by_comment
+        {
+            bool operator()(post_cref lhs, post_cref rhs) const
+            {
+                return lhs.get().comment < rhs.get().comment;
+            }
+        };
+
+        std::set<post_cref, less_by_comment> exclude_set;
+        for (const auto& posts : exclude_posts_by_tags)
+            boost::copy(posts, std::inserter(exclude_set, end(exclude_set)));
+
+        posts_crefs result;
+        result.reserve(filtering_posts.size());
+
+        std::copy_if(begin(filtering_posts), end(filtering_posts), std::back_inserter(result),
```

Comment:
Use boost::set_difference

Reply:
set_difference require that containers must be sorted.
But there is different sorting order for filtering_posts and exclude_set and almost no profit